access/types/stdasync.py

```diff
@@ -390,58 +361,44 @@ def _acceptSubmission(request, course, exercise, post_url, sdir):
-        }
-        if exercise.get("personalized", False):
-            exercise_extra["personalized_exercise"] \
-                = select_generated_exercise_instance(course, exercise, uids, attempt)
-            if settings.ENABLE_PERSONAL_DIRECTORIES:
-                exercise_extra["personal_directory"] \
-                    = user_personal_directory_path(course, exercise, uids)
-
-        sid = os.path.basename(sdir)
-        write_submission_meta(sid, {
-            "url": surl,
-            "dir": sdir,
-            "course_key": course["key"],
-            "exercise_key": exercise["key"],
-            "lang": translation.get_language(),
-        })
-        r = invoke([
-            settings.CONTAINER_SCRIPT,
-            sid,
-            request.scheme + "://" + request.get_host(),
-            c["image"],
-            os.path.join(DIR, course["key"], c["mount"]),
-            sdir,
-            c["cmd"],
-            json.dumps(course_extra),
-            json.dumps(exercise_extra),
-        ])
-        LOGGER.debug("Container order exit=%d out=%s err=%s",
-            r["code"], r["out"], r["err"])
-        qlen = 1
-
-    # Queue in celery & rabbitmq for chroot sandbox actions.
-    else:
-        tasks.grade.delay(course["key"], exercise["key"],
-            translation.get_language(), surl, sdir, uids, attempt)
-        qlen = queue_length()
-        LOGGER.debug("Submission of %s/%s, queue counter %d, queue length %d",
-            course["key"], exercise["key"], _acceptSubmission.counter, qlen)
-        if qlen >= settings.QUEUE_ALERT_LENGTH:
-            LOGGER.error("Queue alert, length: %d", qlen)
+    c = _requireContainer(exercise)
+
+    course_extra = {
+        "key": course["key"],
+        "name": course["name"],
+    }
+    exercise_extra = {
+        "key": exercise["key"],
+        "title": exercise.get("title", None),
+        "resources": c.get("resources", {}), # Unofficial param, implemented differently later
+        "require_constant_environment": c.get("require_constant_environment", False) # Unofficial param, implemented differently later
+    }
+    if exercise.get("personalized", False):
+        exercise_extra["personalized_exercise"] \
+            = select_generated_exercise_instance(course, exercise, uids, attempt)
+
+    sid = os.path.basename(sdir)
+    write_submission_meta(sid, {
+        "url": surl,
+        "dir": sdir,
+        "course_key": course["key"],
+        "exercise_key": exercise["key"],
+        "lang": translation.get_language(),
+    })
+    r = invoke([
+        settings.CONTAINER_SCRIPT,
+        sid,
+        request.scheme + "://" + request.get_host(),
+        c["image"],
+        os.path.join(DIR, course["key"], c["mount"]),
+        sdir,
+        c["cmd"],
+        json.dumps(course_extra),
+        json.dumps(exercise_extra),
+    ])
+    LOGGER.debug("Container order exit=%d out=%s err=%s",
+        r["code"], r["out"], r["err"])
+    qlen = 1
```

Comment:
Is the `qlen` variable used for anything anymore or could it be removed?

Reply:
Wait, I see it in the template context below. Though, the value is always one, but this is not very important to change right now. It is basically not used, but the template is old.